docs/releases/5.2.txt

```diff
@@ -297,6 +309,11 @@ Models
 * Added support for validation of model constraints which use a
   :class:`~django.db.models.GeneratedField`.
 
+* Introduced ``AutoFieldMixin` to decouple the automatic primary key generation
+  behavior from the underlying data type. This allows creating
+  auto-incrementing primary keys of any data type, such as UUID fields,
+  without being limited to integer-based primary keys with ``AutoField``.
```

Comment:
I feel like this is a useful change on its own that would unlock other kinds of autofields as 3rd party package. As this PR has stalled a bit, is it worth extracting it?

Reply:
+1 for this. I can try to help with extracting it if help is needed.

Reply:
I will align with upstream main in a moment.